From 16995239034d44c1dbe1e5f8cf2bd546f2a8d5d3 Mon Sep 17 00:00:00 2001
From: Tommy Huang <tommy_huang@aspeedtech.com>
Date: Tue, 16 Sep 2025 19:23:01 +0800
Subject: [PATCH 07/10] drivers: aspeed_mbox: add invalid d-cache in the mbox
 read api

Add invalid d-cache behavior in the mailbox read api.

Signed-off-by: Tommy Huang <tommy_huang@aspeedtech.com>
Change-Id: I953519280ba298bc4fd7365291d5b47231a36af8
(cherry picked from commit b36542bd381386f0542263d621bdacfe272d61fd)
Signed-off-by: Vasanthakumar K <vasanthakumark@ami.com>
---
 drivers/misc/aspeed_mbox.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/misc/aspeed_mbox.c b/drivers/misc/aspeed_mbox.c
index ec6a133d33..44dba648ba 100644
--- a/drivers/misc/aspeed_mbox.c
+++ b/drivers/misc/aspeed_mbox.c
@@ -47,6 +47,10 @@ static int aspeed_mbox_read(struct udevice *dev, int offset,
 		return -EINVAL;
 	}
 
+	/* invalidate dcache before memory copy */
+	invalidate_dcache_range((ulong)mb->rx_base.buf,
+				(ulong)mb->rx_base.buf + mb->rx_base.size);
+
 	memcpy(buf, mb->rx_base.buf + offset, size);
 	return size;
 }
-- 
2.34.1

